File: app/src/main/java/de/tum/in/tumcampusapp/ui/tufilm/FilmCard.java
package de.tum.in.tumcampusapp.ui.tufilm;

import android.content.Context;
import android.content.Intent;
import androidx.annotation.Nullable;

import de.tum.in.tumcampusapp.navigation.NavigationDestination;
import de.tum.in.tumcampusapp.navigation.SystemIntent;
import de.tum.in.tumcampusapp.ui.news.NewsCard;
import de.tum.in.tumcampusapp.ui.overview.CardManager;
import de.tum.in.tumcampusapp.core.Const;
import de.tum.in.tumcampusapp.core.DateTimeUtils;

public class FilmCard extends NewsCard {

    public FilmCard(Context context) {
        super(CardManager.CARD_NEWS_FILM, context);
    }

    @Nullable
    @Override
    public NavigationDestination getNavigationDestination() {
        Intent intent = new Intent(getContext(), KinoActivity.class);
        intent.putExtra(Const.KINO_DATE, DateTimeUtils.INSTANCE.getDateTimeString(getDate()));
        return new SystemIntent(intent);
    }

}

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/tufilm/FilmCard.kt
package de.tum.`in`.tumcampusapp.component.ui.tufilm

import android.content.Context
import android.content.Intent

import de.tum.`in`.tumcampusapp.component.other.navigation.NavigationDestination
import de.tum.`in`.tumcampusapp.component.other.navigation.SystemIntent
import de.tum.`in`.tumcampusapp.component.ui.news.NewsCard
import de.tum.`in`.tumcampusapp.component.ui.news.model.News
import de.tum.`in`.tumcampusapp.component.ui.overview.CardManager
import de.tum.`in`.tumcampusapp.utils.Const
import de.tum.`in`.tumcampusapp.utils.DateTimeUtils

class FilmCard(context: Context, news: News) : NewsCard(CardManager.CARD_NEWS_FILM, context, news) {
    override fun getNavigationDestination(): NavigationDestination? {
        val intent = Intent(context, KinoActivity::class.java).apply {
            putExtra(Const.KINO_DATE, DateTimeUtils.getDateTimeString(date))
        }
        return SystemIntent(intent)
    }
}
